src/engraving/dom/measure.cpp

```diff
@@ -3108,6 +3116,41 @@ EngravingItem* Measure::prevElementStaff(staff_idx_t staff)
     return score()->firstElement();
 }
 
+bool Measure::hasCrossStaffOrModifiedBeams() const
+{
+    for (const Segment& seg : m_segments) {
+        if (!seg.isChordRestType()) {
+            continue;
+        }
+        for (EngravingItem* e : seg.elist()) {
+            if (!e || !e->isChordRest()) {
+                continue;
+            }
+            if (toChordRest(e)->beam() && (toChordRest(e)->beam()->cross() || toChordRest(e)->beam()->userModified())) {
```

Comment:
It would be better to write the result of toChordRest(e)->beam() to a temp variable and reuse it